src/routes/group.route.ts

```diff
@@ -0,0 +1,27 @@
+import { Router } from 'express';
+import GroupsController from '@controllers/groupe.controller';
+import { CreateGroupDto } from '@dtos/group.dto';
+import { Routes } from '@interfaces/routes.interface';
+import validationMiddleware from '@middlewares/validation.middleware';
+
+class GroupsRoute implements Routes {
+  public path = '/groups';
+  public router = Router();
+  public groupsController = new GroupsController();
+
+  constructor() {
+    this.initializeRoutes();
+  }
```

Comment:
nhance encapsulation by making internal variables private to maintain the principle of information hiding. Additionally, consider passing objects through the constructor to maintain decoupling and promote modular design practices. This approach fosters code integrity and supports easier maintenance in the long run.

```suggestion
  private path = '/groups';
  private router = Router();

  constructor(private groupsController: GroupsController) {
    this.initializeRoutes();
  }
```